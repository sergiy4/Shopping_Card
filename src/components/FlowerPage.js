import { useSpring, animated } from '@react-spring/web'
import Chinese from '../img/png/aglonema.png'
import '../styles/PlantPage.css' 

function FlowerPage(props){

    const {imgPlant,name,price} = props
    
    const [itemStyle, setItemStyle] = useSpring(() => ({boxShadow:'0px 0px 0px 0px rgb(0, 0, 0)',top:'0%',left:'0%'}));
    const [BtnStyle,setBtnStyle] = useSpring(()=> ({boxShadow:'0px 0px 0px 0px rgb(0, 0, 0)',top:'0%',left:'0%'}))
    
    const [textStyle, set] = useSpring(() => ({textShadow:'0px 0px 0px rgb(0, 0, 0)',top:'0%',left:'0%'}));
    const handleMouseEnter = () =>{
        setItemStyle({
            boxShadow:'15px 15px 0px 0px rgb(0, 0, 0)',
            top:'-2.7%',
            left:'-2.7%' 
        })
    }

    const handleMouseLeave = ()=>{
        setItemStyle({
            boxShadow:'0px 0px 0px 0px rgb(0, 0, 0)',
            top:'0%',
            left:'0%'
        })
    }
    return(

       <main className='plant_main'>
            <div className='Around_item'
                onMouseEnter={handleMouseEnter}
                 
                onMouseLeave={handleMouseLeave}
            >
            
                <animated.div   className='item' style={itemStyle}> 
                
                    <img src={Chinese} alt='plant' className='img_plant'></img>
                    <div className='Info'>
                        <div className='right_part'>
                            <animated.h1
                                style={textStyle} 
                                className='text'
                            >sdfsdfdsfsd
                            </animated.h1>

                            <animated.h3
                                style={textStyle}
                                className='text'
                            >sdfsddfsdf
                            </animated.h3>

                            <div className='around_btn'
                                onMouseEnter={()=>{
                                    setBtnStyle({boxShadow:'10px 10px 0px 0px rgb(0, 0, 0)',top:'-2.7%',left:'-2.7%'})
                                    set({textShadow:'3px 2px 0px rgb(0, 0, 0)',top:'-2.7%',left:'-2.7%'})
                                }}   
                                onMouseLeave={()=>{
                                    setBtnStyle({boxShadow:'0px 0px 0px 0px rgb(0, 0, 0)',top:'0%',left:'0%'})
                                    set({textShadow:'0px 0px 0px rgb(0, 0, 0)',top:'0%',left:'0%'})
                                }}
                            >
                                <animated.button  
                                    style={BtnStyle}
                                    className='btn_buy'
                                    >BUY</animated.button>
                            </div>
                        </div>
                    </div>
                </animated.div>
            </div>
          
       </main>

    )
}
export default FlowerPage