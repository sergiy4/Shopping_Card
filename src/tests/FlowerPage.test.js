import {render, screen,cleanup} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import "@testing-library/jest-dom"
import FlowerPage from '../components/FlowerPage'

afterEach(()=>{
    cleanup()
})

describe('Testing FlowerPage',()=>{
    test('correctly render FlowerPage',()=>{
        const {asFragment} = render(<FlowerPage/>)

        expect(asFragment()).toMatchSnapshot()
        
    }),
    test('renders img and two buttons',()=>{
        render(<FlowerPage/>)

        const img = screen.getByRole('img',{name:'plant'})
        const button = screen.getByRole('button')

        expect(img).toBeInTheDocument()
        expect(button).toBeInTheDocument()
        expect(img).toHaveClass('img_plant')
    })


})
