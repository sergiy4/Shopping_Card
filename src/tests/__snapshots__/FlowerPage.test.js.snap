// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FlowerPage correctly render FlowerPage 1`] = `
<DocumentFragment>
  <main
    class="flower_main"
  >
    <div
      class="left_part"
    >
      <img
        alt="plant"
        class="img_plant"
        src="AsparagusFern.jpg"
      />
    </div>
    <div
      class="right_part"
    >
      <h1 />
      <h3 />
      <button>
        BUY
      </button>
    </div>
  </main>
</DocumentFragment>
`;
